domain/relation/service/relation.go

```diff
@@ -478,11 +538,272 @@ func (s *WatchableService) WatchApplicationSettings(
 
 // WatchLifeSuspendedStatus returns a watcher that notifies of changes to the life
 // or suspended status of the relation.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.ApplicationIDNotValid] is returned if the application
+//     is not valid.
 func (s *WatchableService) WatchLifeSuspendedStatus(
+	applicationID application.ID,
+) (watcher.StringsWatcher, error) {
+	if err := applicationID.Validate(); err != nil {
+		return nil, errors.Errorf(
+			"%w: %w", relationerrors.ApplicationIDNotValid, err)
+	}
+	currentRelations := make(map[corerelation.UUID]relation.RelationLifeSuspendedData)
+	// returns a set of relation keys if the life or suspended status has changed
+	// for any relation this application is part of.
+
+	lifeTableName, suspendedTableName, query := s.st.InitialWatchLifeSuspendedStatus(applicationID)
+	return s.watcherFactory.NewNamespaceMapperWatcher(
+		func(ctx context.Context, txn database.TxnRunner) ([]string, error) {
+			relationUUIDStrings, err := query(ctx, txn)
+			if err != nil {
+				return nil, errors.Capture(err)
+			}
+
+			initialResults := make([]string, len(relationUUIDStrings))
+
+			for i, relUUID := range relationUUIDStrings {
+				relUUID := corerelation.UUID(relUUID)
+				relationData, err := s.st.WatchLifeSuspendedStatusMapperData(ctx, relUUID, applicationID)
+				if errors.Is(err, relationerrors.ApplicationNotFoundForRelation) {
+					continue
+				} else if err != nil {
+					return nil, errors.Capture(err)
+				}
+				currentRelations[relUUID] = relationData
+				initialResults[i] = relationData.Key.String()
+			}
+
+			return initialResults, nil
+		},
+		func(ctx context.Context, txn database.TxnRunner, changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+			// If there are no changes, return no changes.
+			if len(changes) == 0 {
+				return nil, nil
+			}
+			var err error
+			changeEvents := []changestream.ChangeEvent{}
+			changeEvents, currentRelations, err = s.changeEventsForLifeSuspendedStatusMapper(
+				ctx,
+				applicationID,
+				changeEvents,
+				currentRelations,
+			)
+			if err != nil {
+				return nil, errors.Capture(err)
+			}
+			return changeEvents, nil
+		},
+		eventsource.NamespaceFilter(lifeTableName, changestream.All),
+		eventsource.NamespaceFilter(suspendedTableName, changestream.All),
+	)
+}
+
+func (s *WatchableService) changeEventsForLifeSuspendedStatusMapper(
 	ctx context.Context,
-	relationUUID corerelation.UUID,
+	applicationID application.ID,
+	changes []changestream.ChangeEvent,
+	currentRelations map[corerelation.UUID]relation.RelationLifeSuspendedData,
+) ([]changestream.ChangeEvent, map[corerelation.UUID]relation.RelationLifeSuspendedData, error) {
+
+	changeEvents := make([]changestream.ChangeEvent, 0)
+
+	for _, change := range changes {
+		relUUID := corerelation.UUID(change.Changed())
+
+		changedRelationData, err := s.st.WatchLifeSuspendedStatusMapperData(ctx, relUUID, applicationID)
```

Comment:
```suggestion
		changedRelationData, err := s.st.WatchLifeSuspendedStatusMapperData(ctx, relUUID, applicationID)
```

That's starting to read like JavaBeanFactoryVistor...